cpp/ql/test/query-tests/Critical/MemoryFreed/test.cpp

```diff
@@ -198,3 +198,9 @@ void test_strndupa_dealloc() {
 	char *cpy = strndupa(msg, 4);
     free(cpy); // BAD [NOT DETECTED]
 }
+
+void test_free_malloc() {
+	void *a = malloc(10);
+	void *b;
+	free(b = a);
```

Comment:
I think this needs some kind of `// GOOD` or `// BAD` comment. However, it's not clear to me to which query being tested here the comment applies.